core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -596,19 +615,31 @@ private void innerDelete(Delete delete) throws IOException {
             }
 
             final long expectedVersion = delete.version();
-            if (checkVersionConflict(delete, currentVersion, expectedVersion, deleted)) return;
-
-            final long updatedVersion = updateVersion(delete, currentVersion, expectedVersion);
-
-            final boolean found = deleteIfFound(delete, currentVersion, deleted, versionValue);
-
-            delete.updateVersion(updatedVersion, found);
-
-            maybeAddToTranslog(delete, updatedVersion, Translog.Delete::new, DeleteVersionValue::new);
+            if (checkVersionConflict(delete, currentVersion, expectedVersion, deleted)) {
+                // skip executing delete because of version conflict on recovery
+                return new DeleteResult(expectedVersion, true,
```

Comment:
weird line wrap + the same request as before - one return statement (with a freeze)